test/e2e/run-openrpc-api-test-coverage.ts

```diff
@@ -0,0 +1,644 @@
+const mockServer = require('@open-rpc/mock-server/build/index').default;
+import HtmlReporter from '@open-rpc/test-coverage/build/reporters/html-reporter';
+import ExamplesRule from '@open-rpc/test-coverage/build/rules/examples-rule';
+import JsonSchemaFakerRule from '@open-rpc/test-coverage/build/rules/json-schema-faker-rule';
+import paramsToObj from '@open-rpc/test-coverage/build/utils/params-to-obj';
+import {
+  ContentDescriptorObject,
+  ExampleObject,
+  ExamplePairingObject,
+  MethodObject,
+  OpenrpcDocument,
+} from '@open-rpc/meta-schema';
+import openrpcDocument from '@metamask/api-specs';
+const { v4 } = require('uuid');
+
+declare let window: any;
+
+const uuid = v4;
+
+const FixtureBuilder = require('./fixture-builder');
+const {
+  withFixtures,
+  openDapp,
+  unlockWallet,
+  DAPP_URL,
+  WINDOW_TITLES,
+  switchToOrOpenDapp,
+} = require('./helpers');
+
+import { Driver, PAGES } from './webdriver/driver';
+import Rule from '@open-rpc/test-coverage/build/rules/rule';
+import { Call } from '@open-rpc/test-coverage/build/coverage';
+
+const pollForResult = async (driver: Driver, generatedKey: string) => {
+  let result = await driver.executeScript(`return window['${generatedKey}'];`);
+
+  while (result === null) {
+    // Continue polling if result is not set
+    await driver.delay(50);
+    result = await driver.executeScript(`return window['${generatedKey}'];`);
+  }
+
+  // clear the result
+  await driver.executeScript(`delete window['${generatedKey}'];`);
+
+  return result;
+};
+
+const createDriverTransport = (driver: Driver) => {
+  return async (
+    _: string,
+    method: string,
+    params: any[] | Record<string, any>,
+  ) => {
+    const generatedKey = uuid();
+    // don't wait for executeScript to finish window.ethereum promise
+    // we need this because if we wait for the promise to resolve it
+    // will hang in selenium since it can only do one thing at a time.
+    // the workaround is to put the response on window.asyncResult and poll for it.
+    driver.executeScript(
+      ([m, p, g]: any) => {
+        window[g] = null;
+        window.ethereum
+          .request({ method: m, params: p })
+          .then((r: any) => {
+            window[g] = { result: r };
+          })
+          .catch((e: any) => {
+            window[g] = {
+              error: {
+                code: e.code,
+                message: e.message,
+                data: e.data,
+              },
+            };
+          });
+      },
+      method,
+      params,
+      generatedKey,
+    );
+    const response = await pollForResult(driver, generatedKey);
+    return response;
+  };
+};
+
+interface ConfirmationsRejectRuleOptions {
+  driver: Driver;
+  only: string[];
+}
+// this rule makes sure that all confirmation requests are rejected.
+// it also validates that the JSON-RPC response is an error with
+// error code 4001 (user rejected request)
+class ConfirmationsRejectRule implements Rule {
```

Comment:
Maybe we make a new `ApiTestingRules.js` (or something like that) file for these?